node/src/client/router.rs

```diff
@@ -48,7 +48,10 @@ impl<N: Network, C: ConsensusStorage<N>> Handshake for Client<N, C> {
         let conn_side = connection.side();
         let stream = self.borrow_stream(&mut connection);
         let genesis_header = *self.genesis.header();
-        let (peer_ip, mut framed) = self.router.handshake(peer_addr, stream, conn_side, genesis_header).await?;
+        let (peer, mut framed) = self.router.handshake(peer_addr, stream, conn_side, genesis_header).await?;
+        let peer_ip = peer.ip();
+
+        self.router.insert_connected_peer(peer, peer_addr);
```

Comment:
this setup is not "tight" enough, as the list of connecting peers would not be updated (or rather cleaned up) if the call to `Router::handshake` failed; note how it is done originally - the list of connecting peers is always updated regardless of the result of the handshake

Reply:
Good catch, will correct 👍